### PR TITLE
Custom Taco & Kebabs - FoodSequence 2

### PR DESCRIPTION
**Pull request by TheShuEd**
## About the PR
- removed all microwave taco & kebabs recipes (except for the taco shell itself)
- added morbilliarb variants of procedural tacos and kebabs
- improved control over the location and order of food layers generated from FoodSequence
  - we can invert the rendering of the layers
  - we can place generated layers in front of or behind static layers
- fixed some bugs in the name generator
- added possibility to specify a base sprite for any states
- mice and hamlet can now be put in tacos and burgers (must be dead)

EDIT:
- added layer offset randomization
- you can fight with a skewer (even if it has food on it)
- now you can't put more than 10 layers on a burger.

## Media
image
image
image

**Changelog**

:cl:
- add: added morbilliard variants of procedural tacos and kebabs
- remove: removed all microwave taco and kebabs recipes (except for the taco shell itself)
- tweak: you can fight with a skewer (even if it has food on it)
- tweak: now you can't put more than 10 layers on a burger. (20 before)

### CODE REVIEW

**Comment by TheShuEd** [comment]
image
offset randomization

**Comment by UbaserB** [comment]
The offset feels too strong imo

**Comment by TheShuEd** [comment]
image

**Comment by deltanedas** [comment]
chicken not chiken :trollface:

**Comment by TheShuEd** [comment]
fixed already

**Comment by Pumkin69** [comment]
these prs are fucking amazing

**Comment by TheShuEd** [comment]
> hey uh question , can lizards ect also eat those if its just meat? problem with burger was even if it was fully meats they couldnt eat it or a fruity burger

I'm going to fix it in the next PR

File: Resources/Prototypes/Entities/Objects/Consumable/Food/burger.yml [hunk 1]
```diff
@@ -43,7 +43,7 @@
     - map: ["foodSequenceLayers"]
   - type: FoodSequenceStartPoint
     key: burger
-    maxLayers: 20
+    maxLayers: 4
```
Comment (deltanedas) [inline Resources/Prototypes/Entities/Objects/Consumable/Food/burger.yml]: literally 1984 why make it so low now

Comment (TheShuEd) [reply Resources/Prototypes/Entities/Objects/Consumable/Food/burger.yml]: image

Comment (deltanedas) [reply Resources/Prototypes/Entities/Objects/Consumable/Food/burger.yml]: still

Comment (TheShuEd) [reply Resources/Prototypes/Entities/Objects/Consumable/Food/burger.yml]: image

Comment (deltanedas) [reply Resources/Prototypes/Entities/Objects/Consumable/Food/burger.yml]: so make them increase in size, 20 ingredients = 20 tall burger (impossible to store except in boh)

Comment (TheShuEd) [reply Resources/Prototypes/Entities/Objects/Consumable/Food/burger.yml]: We don't have a normal functionality for editing item size yet, and adding it is not within the scope of this PR.

File: Resources/Prototypes/Entities/Objects/Consumable/Food/meat.yml [hunk 2]
```diff
@@ -823,12 +882,12 @@
...
   - type: FoodSequenceElement
+    sprite:
+      sprite: Objects/Consumable/Food/meat.rsi
+      state: meatball
     entries:
       burger:
         name: food-sequence-burger-content-raw-meat
-        sprite:
-          sprite: Objects/Consumable/Food/meat.rsi
-          state: meatball
```
Comment (slarticodefast) [inline Resources/Prototypes/Entities/Objects/Consumable/Food/meat.yml:831]: no meatball taco?

File: Resources/Prototypes/Entities/Objects/Consumable/Food/meat.yml [hunk 3]
```diff
@@ -1233,12 +1339,14 @@
...
+    sprite:
+      sprite: Objects/Consumable/Food/meat.rsi
+      state: rouny-cooked
     entries:
       burger:
-        name: food-sequence-burger-content-xeno
-        sprite:
-          sprite: Objects/Consumable/Food/meat.rsi
-          state: rouny-cooked
+        name: food-sequence-content-xeno
```
Comment (slarticodefast) [inline Resources/Prototypes/Entities/Objects/Consumable/Food/meat.yml:1347]: same as above

File: Resources/Prototypes/Entities/Objects/Consumable/Food/ingredients.yml [hunk 4]
```diff
@@ -600,12 +593,14 @@
...
+      state: cheesewedge
     entries:
       burger:
-        name: food-sequence-burger-content-cheese
-        sprite:
-          sprite: Objects/Consumable/Food/ingredients.rsi
-          state: cheesewedge
+        name: food-sequence-content-cheese
+      taco: 
+        name: food-sequence-content-cheese
```
Comment (slarticodefast) [inline Resources/Prototypes/Entities/Objects/Consumable/Food/ingredients.yml:603]: You should add FoodSequenceElement to tofu and the other cheese.

File: Resources/Prototypes/Entities/Mobs/NPCs/animals.yml [hunk 5]
```diff
@@ -1675,6 +1675,21 @@
   - type: FireVisuals
     sprite: Mobs/Effects/onfire.rsi
     normalState: Mouse_burning
+  - type: FoodSequenceElement
+    sprite:
+      sprite: Mobs/Animals/mouse.rsi
+      state: dead-0
+    entries:
+      burger:
+        name: food-sequence-burger-content-rat
```
Comment (slarticodefast) [inline Resources/Prototypes/Entities/Mobs/NPCs/animals.yml:1684]: maybe distinguish between mice and rats?

**Review by slarticodefast — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Entities/Objects/Consumable/Food/produce.yml [hunk 6]
```diff
@@ -876,12 +898,14 @@
...
+    sprite:
+      sprite: Objects/Specific/Hydroponics/blue_tomato.rsi
+      state: produce
     entries:
       burger:
-        name: food-sequence-burger-content-tomato
-        sprite:
-          sprite: Objects/Specific/Hydroponics/blue_tomato.rsi
-          state: produce
+        name: food-sequence-content-tomato
```
[the remark this answers is not in the document]
Comment (TheShuEd) [reply Resources/Prototypes/Entities/Objects/Consumable/Food/produce.yml:906]: yep, later, i dont make sprites in this PR

**Review by slarticodefast — APPROVED** [review]

File: Resources/Prototypes/Entities/Objects/Consumable/Food/burger.yml (hunk 1 above)
Comment (Everturning) [reply Resources/Prototypes/Entities/Objects/Consumable/Food/burger.yml]: we used to

**Pull request merged**